modules/listener/listener.bal

```diff
@@ -0,0 +1,154 @@
+        return self.httpListener.attach(s, name);
+    }
+
+    public isolated function detach(http:Service s) returns error? {
+        return self.httpListener.detach(s);
+    }
+
+    public isolated function 'start() returns error? {
+        return self.httpListener.'start();
+    }
+
+    public isolated function gracefulStop() returns error? {
+        return ();
+    }
+
+    public isolated function immediateStop() returns error? {
+        return self.httpListener.immediateStop();
+    }
+
+    # Finding event type triggered and retrieve changes list.
+    # 
+    # + caller - The http caller object for responding to requests 
+    # + request - The HTTP request.
+    # + return - Returns error, if unsuccessful.
+    public function findEventType(http:Caller caller, http:Request request) returns @tainted error? {
+        log:print("< RECEIVING A CALLBACK <");
+        string channelID = check request.getHeader("X-Goog-Channel-ID");
+        string messageNumber = check request.getHeader("X-Goog-Message-Number");
+        string resourceStates = check request.getHeader("X-Goog-Resource-State");
+        string channelExpiration = check request.getHeader("X-Goog-Channel-Expiration");
+        if (channelID != self.channelUuid) {
+            fail error("Diffrent channel IDs found, Resend the watch request");
+        } else {
+            drive:ChangesListResponse[] response = check getAllChangeList(self.currentToken, self.driveClient);
+            foreach drive:ChangesListResponse item in response {
+                self.currentToken = item?.newStartPageToken.toString();
+                if (self.isWatchOnSpecificResource && self.isAFolder) {
+                    log:print("Folder watch response processing");
+                    check mapEventForSpecificResource(<@untainted> self.specificFolderOrFileId, <@untainted> item, 
+                    <@untainted> self.driveClient, <@untainted> self.eventService, <@untainted> self.currentFileStatus);
+                    check getCurrentStatusOfDrive(self.driveClient, self.currentFileStatus, self.specificFolderOrFileId);
+                } else if (self.isWatchOnSpecificResource && self.isAFolder == false) {
+                    log:print("File watch response processing");
+                    check mapFileUpdateEvents(self.specificFolderOrFileId, item, self.driveClient, self.eventService, 
+                    self.currentFileStatus);
+                    check getCurrentStatusOfFile(self.driveClient, self.currentFileStatus, self.specificFolderOrFileId);
+                } else {
+                    log:print("Whole drive watch response processing");
+                    check mapEvents(item, self.driveClient, self.eventService, self.currentFileStatus);
+                    check getCurrentStatusOfDrive(self.driveClient, self.currentFileStatus);
+                }
+            }
+        }
+        log:print("< RECEIVED >");
```

Comment:
Do we need log lines like this? What does this log means?

Reply:
Removed the unwanted log lines in all occurences